src/mode/modeHandler.ts

```diff
@@ -129,6 +131,40 @@ export class VimSettings {
   useCtrlKeys     = false;
 }
 
+export class ComposeState {
```

Comment:
Then I think you might be okay with just using some closure to handle the sharing of `content` and `isComposing`, right?